src/controller/CHIPDeviceController.cpp

```diff
@@ -118,6 +118,9 @@ CHIP_ERROR ChipDeviceController::Init(NodeId localNodeId)
     err = DeviceLayer::PlatformMgr().InitChipStack();
     SuccessOrExit(err);
 
+    err = Platform::MemoryInit();
```

Comment:
Should this happen before the CHIP Stack is init?

Reply:
It's happing after `InitChipStack` in all other places.